src/Microsoft.SqlTools.ServiceLayer/QueryExecution/SpecialAction.cs

```diff
@@ -0,0 +1,132 @@
+// 
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+namespace Microsoft.SqlTools.ServiceLayer.QueryExecution
+{
+    /// <summary>
+    /// Class that represents a Special Action which occured by user request during the query 
+    /// </summary>
+    public class SpecialAction {
+        
+        #region Private Class variables 
+        bool none;
+        bool expectActualYukonXmlShowPlan;
+        bool expectEstimatedYukonXmlShowPlan;
+
+        #endregion
+
+        /// <summary>
+        /// a type of XML execution plan may be returned  
+        /// </summary>
+        public SpecialAction()
+        {
+            None = true;
+            ExpectActualYukonXmlShowPlan = false;
+            ExpectEstimatedYukonXmlShowPlan = false;
+        }
+
+        #region Public Functions
+        /// <summary>
+        /// No Special action performed 
+        /// </summary>
+        public bool None {
+            get { return none; }
+            set { 
+                none = value;
+                if (value)
+                {
+                    expectActualYukonXmlShowPlan = false;
+                    expectEstimatedYukonXmlShowPlan = false;
+                }
+            }
+        }
+
+        /// <summary>
+        /// Actual XML execution plan may be returned 
+        /// </summary>
+        public bool ExpectActualYukonXmlShowPlan 
+        {
+            get { return expectActualYukonXmlShowPlan; }
+            set { this.registerSpecialAction(ref expectActualYukonXmlShowPlan, value); }
+        }
+
+        /// <summary>
+        /// Estimated XML execution plan may be returned  
+        /// </summary>
+        public bool ExpectEstimatedYukonXmlShowPlan 
+        {
+            get { return expectEstimatedYukonXmlShowPlan; }
+            set { this.registerSpecialAction(ref expectEstimatedYukonXmlShowPlan, value); }
+        }
+
+        /// <summary>
+        /// a type of XML execution plan may be returned  
+        /// </summary>
+        public bool ExpectYukonXMLShowPlan 
+        {
+            get { return ExpectEstimatedYukonXmlShowPlan || ExpectActualYukonXmlShowPlan; }
+            set 
+            { 
+                ExpectEstimatedYukonXmlShowPlan = value;
+                ExpectActualYukonXmlShowPlan = value;
```

Comment:
When it is set, it basically lets us know that there is an XML execution plan that will be returned, but we are not sure which one. It is not possible to know which kind of plan is returned unless I couple the query execution options to the result set summaries (i.e. Save a copy of the QueryExecutionSettings in the resultSet object to later refer to when when setting up the special action). I am not sure exactly what to do in this case.